core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2788,6 +2788,18 @@ public static double log(BigDecimal d0, BigDecimal d1) {
     return Math.log(d0.doubleValue()) / Math.log(d1.doubleValue());
   }
 
+  /** SQL {@code LOG2(number)} function applied to double values. */
+  public static double log2(double d0) {
+    return (d0 == 0) ? Double.NaN : Math.log(d0) / Math.log(2);
```

Comment:
In general testing for equality FP values does not work.
There are also two kinds of zeros in FP, +0 and -0.
If the result from Math.log(0) is -Infinity, the more robust way to do this is to actually compute the log and check when the result is the negative infinity.

Reply:
Thank you, learned it